pepper-apis/src/test/java/org/broadinstitute/ddp/script/InsertInvitationRecruitmentCodesScript.java

```diff
@@ -17,33 +29,96 @@
+                        .filter(badWord -> newCode.contains(badWord)).findAny();
+                if (containedBadWord.isEmpty()) {
+                    InvitationDto invitation = insertRecruitmentInvitationCode(newCode, study, handle);
+                    ++insertedCount;
+                    LOG.info("Invitation code: {} inserted with id {}. {} left", newCode, invitation.getInvitationId(),
+                            (qty - insertedCount));
+                } else {
+                    LOG.warn("Skipping inserting code: {} because it contains the bad word: {}", newCode, containedBadWord.get());
+                }
+            } while (insertedCount < qty);
         });
     }
 
+    private List<String> readAllPossibleBadWords(String directoryPath) {
+        List<File> wordFiles;
+        try {
+            wordFiles = Files.list(Paths.get(directoryPath))
+                    .map(path -> path.toFile())
+                    .filter(file -> !file.getName().contains("."))
+                    .filter(file -> file.getName().toLowerCase().equals(file.getName())) // ignore files with uppercase chars in name
+                    .collect(toList());
+        } catch (IOException e) {
+            throw new DDPException("Could not read the bad word directory at:" + directoryPath, e);
+        }
+        List<String> badWordList = wordFiles.stream().flatMap(wordFile -> {
+            try {
+                LOG.info("Reading bad word file: {}", wordFile.getAbsolutePath());
+                return ((List<String>) (FileUtils.readLines(wordFile, "UTF-8"))).stream()
+                        .filter(word -> StringUtils.isNotBlank(word));
+            } catch (IOException e) {
+                throw new DDPException("Could not read bad word file:" + wordFile.getAbsolutePath(), e);
+            }
+        }).collect(toList());
+
+        return badWordList.stream().map(word -> normalizeWord(word)).filter(word -> isWordPossible(word)).collect(toList());
+    }
+
+    private String normalizeWord(String word) {
+        return word.toUpperCase().replaceAll("\\s", "");
+    }
+
+    private boolean isWordPossible(String normalizedWord) {
+        if (normalizedWord.length() > CODE_LENGTH) {
+            return false;
+        }
+        Set<Character> wordChars = normalizedWord.codePoints().mapToObj(c -> (char) c).collect(toSet());
+        Set<Character> validChars = new String(GuidUtils.UPPER_ALPHA_NUMERIC_EXCLUDING_CONFUSING_CHAR).codePoints()
+                .mapToObj(c -> (char) c).collect(toSet());
+        return validChars.containsAll(wordChars);
+    }
+
+
     @Test
     public void generateTestBostonInvitationCodes() {
-        insertRecruitmentInvitationCodes("testboston", "TB", 100);
+        insertRecruitmentInvitationCodes("testboston", "TB", 100, "/Users/mocana/Documents/DSP/ddp-study-server/pepper-apis/List-of-Dirty"
+                + "-Naughty-Obscene-and-Otherwise-Bad-Words");
```

Comment:
Might want to reconsider checking in your filesystem hierarchy.

Reply:
6ebfc917885cb5d064f6cb23857f922d7565800d